common/tools.py

```diff
@@ -520,6 +528,56 @@ def checkXServer():
     else:
         return False
 
+
+def is_Qt5_working(systray_required=False):
+    """
+    Check if the Qt5 GUI library is working (installed and configured)
+
+    This function is contained in BiT CLI (not BiT Qt) to allow Qt5
+    diagnostics output even if the BiT Qt GUI is not installed.
+    This function does NOT add a hard Qt5 dependency (just "probing")
+    so it is OK to be in BiT CLI.
+
+    Args:
+        systray_required: Set to ``True`` if the systray of the desktop
+        environment must be available too to consider Qt5 as "working"
+
+    Returns:
+        bool: ``True``  Qt5 can create a GUI
+              ``False`` Qt5 fails (or the systray is not available
+                        if ``systray_required`` is ``True``)
+    """
+
+    # Spawns a new process since it may crash with a SIGABRT and we
+    # don't want to crash BiT if this happens...
+
+    try:
+        path = os.path.join(backintimePath("common"), "qt5_probing.py")
+        cmd = [sys.executable, path]
+        with subprocess.Popen(cmd,
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE,
+                              universal_newlines=True) as proc:
+
+            std_output, error_output = proc.communicate()  # to get the exit code
+
+            logger.debug(f"Qt5 probing result: exit code {proc.returncode}")
+
+            if proc.returncode != 23:  # if some Qt5 parts are missing: Show details
```

Comment:
Damn, I forgot to undo remove my testing code. Will follow up on this with a new commit...

Reply:
I thought it has to be this way. Some more verbose logging in such a complex situation isn't bad.